code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,305 @@
+/datum/action/xeno_action/activable/web_spit
+	name = "Web Spit"
+	ability_name = "Web Spit"
+	mechanics_text = "We spit a stretchy web at our prey"
+	action_icon_state = "web_spit"
+	plasma_cost = 125
+	cooldown_timer = 10 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_WEB_SPIT
+
+/datum/action/xeno_action/activable/web_spit/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/ammo/xeno/web/web_spit = GLOB.ammo_list[/datum/ammo/xeno/web]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(web_spit, web_spit.damage * SPIT_UPGRADE_BONUS(X))
+	newspit.def_zone = X.get_limbzone_target()
+
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Leash Ball
+// ***************************************
+
+/datum/action/xeno_action/activable/leash_ball
+	name = "Leash Ball"
+	ability_name = "Leash Ball"
+	mechanics_text = " Spit a huge ball of web that snares groups of marines "
+	action_icon_state = "leash_ball"
+	plasma_cost = 250
+	cooldown_timer = 20 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+
+/datum/action/xeno_action/activable/leash_ball/use_ability(atom/A)
+	var/turf/target = get_turf(A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(target)
+	if(!do_after(X, 1 SECONDS, TRUE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+
+	var/datum/ammo/xeno/leash_ball = GLOB.ammo_list[/datum/ammo/xeno/leash_ball]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(leash_ball)
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+/obj/structure/xeno/aoe_leash
+	name = "Snaring Web"
+	icon = 'icons/Xeno/Effects.dmi'
+	icon_state = "aoe_leash"
+	desc = "Looks very sticky"
+	destroy_sound = "alien_resin_break"
+	max_integrity = 960
```

Comment:
![image](https://user-images.githubusercontent.com/6610922/189552274-9f9b0ede-962a-4112-b606-2bf3ce80ad99.png)

Resin wall does not have this much integrity. I highly recommend nerfing this.

Reply:
Resin walls on live have up to 300 hp (they regenerate to up to that and 60 bullet / laser armor, not 0 like your screenshot suggests.
This means they have 760 effective health, not 200 as you suggest. This means one snare can only have health equal to at most 1,28 walls worth of health aswell as requiring eight marines snared at once to reach that health.